src/test/ui/type-alias-impl-trait/issue-85113.rs

```diff
@@ -12,6 +12,7 @@ trait Output<'a> {}
 impl<'a> Output<'a> for &'a str {}
 
 fn cool_fn<'a>(arg: &'a str) -> OpaqueOutputImpl<'a> {
```

Comment:
This test is kind of weird. Does it pass if the `let out: ...` is removed?

I expect it to pass, because we stabilized member constraints in https://github.com/rust-lang/rust/pull/84701, but it seems like the `let out: ...` is probably caused it not to work.

Reply:
> This test is kind of weird. Does it pass if the `let out: ...` is removed?

yes, it does. Note that this test has `impl_trait_in_bindings` enabled, so it's not really relevant for min_tait. The problem is that the let binding gives us

> expected `&'<empty> str`, got `&'a str`

which is probably due to the odd way that `impl_trait_in_bindings` works and that should hopefully all go away once we do not treat them special anymore and just put them into the inference list just like we do with the return type.

Reply:
This test is really weird, we were talking about it with @lqd some minutes ago.

Note the following https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=3c4de7d1bfdfee59d8e2f3f5c0d39453

Reply:
I think I'd merge this as is, in my remove let binding PR this test is going away anyway so ...